jenkins/test-image/Makefile

```diff
@@ -12,18 +12,50 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-IMG = gcr.io/google-containers/kubekins-test
+IMG = gcr.io/k8s-testimages/kubekins-test
 TAG = $(shell date +v%Y%m%d)-$(shell git describe --tags --always --dirty)
+FILES := $(wildcard ./Dockerfile-*)
 
-all: build
+do_build = \
+	$(eval FILETAG := $(subst ./Dockerfile-,,$(f)))\
+	docker build -f $(f) -t $(IMG):$(FILETAG)-$(TAG) . ;\
+	docker tag -f $(IMG):$(FILETAG)-$(TAG) $(IMG):$(FILETAG)-latest ;\
+	echo Built $(IMG):$(FILETAG)-$(TAG) and tagged with $(IMG):$(FILETAG)-latest ;
 
-build:
-	docker build -t $(IMG):$(TAG) .
-	docker tag $(IMG):$(TAG) $(IMG):latest
-	@echo Built $(IMG):$(TAG) and tagged with $(IMG):latest
+do_push = \
+	$(eval FILETAG := $(subst ./Dockerfile-,,$(f)))\
+	gcloud docker -- push $(IMG):$(FILETAG)-$(TAG);\
+	gcloud docker -- push $(IMG):$(FILETAG)-latest;\
+	echo Pushed $(IMG):$(FILETAG)-latest and $(IMG):$(FILETAG)-$(TAG);
 
-push: build
-	gcloud docker push $(IMG):$(TAG)
-	gcloud docker push $(IMG):latest
-	@echo Pushed $(IMG):latest and $(IMG):$(TAG)
+.PHONY: all build-all push-all build-single push-single
+
+all:
+	@echo Please Explicitly Specify Your Target
+
+build-all:
+	@$(foreach f,$(FILES),$(do_build))
+
+push-all: build
+	@$(foreach f,$(FILES),$(do_push))
+
+build-single:
+	@$(eval f := ./Dockerfile-$(filter-out $@,$(MAKECMDGOALS)))
+	@$(call do_build,$(f))
+
+push-single:
```

Comment:
I'm guessing you can't just depend on `build-single` here instead of explicitly calling `do_build`?